src/main/java/com/agnieszkapawska/flashcards/domain/facades/LearningFacade.java

```diff
@@ -0,0 +1,122 @@
+package com.agnieszkapawska.flashcards.domain.facades;
+
+import com.agnieszkapawska.flashcards.domain.dtos.FlashcardGetResponseDto;
+import com.agnieszkapawska.flashcards.domain.models.Flashcard;
+import com.agnieszkapawska.flashcards.domain.models.FlashcardsStorage;
+import com.agnieszkapawska.flashcards.domain.models.FlashcardsToLearn;
+import com.agnieszkapawska.flashcards.domain.models.FlashcardsToRefresh;
+import com.agnieszkapawska.flashcards.domain.services.*;
+import com.agnieszkapawska.flashcards.domain.utils.Answer;
+import org.modelmapper.ModelMapper;
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class LearningFacade {
+    private FlashcardsToLearnService flashcardsToLearnService;
+    private FlashcardService flashcardService;
+    private FlashcardsToRepeatService flashcardsToRepeatService;
+    private FlashcardsToRefreshService flashcardsToRefreshService;
+    private ModelMapper modelMapper;
+    private boolean shouldReturnFromMethod;
+
+    public LearningFacade(FlashcardsToLearnService flashcardsToLearnService,
+                          FlashcardService flashcardService,
+                          FlashcardsToRepeatService flashcardsToRepeatService,
+                          FlashcardsToRefreshService flashcardsToRefreshService,
+                          ModelMapper modelMapper) {
+        this.flashcardsToLearnService = flashcardsToLearnService;
+        this.flashcardService = flashcardService;
+        this.flashcardsToRepeatService = flashcardsToRepeatService;
+        this.flashcardsToRefreshService = flashcardsToRefreshService;
+        this.modelMapper = modelMapper;
+    }
+
+    public List<FlashcardGetResponseDto> getFlashcardsToLearnByUserId(Long userId) {
+        FlashcardsToLearn flashcardsToLearn = flashcardsToLearnService.findByUserId(userId).get();
```

Comment:
`flashcardsToLearnService.findByUserId(userId)` is optional and you call `get` on it without checking if it's not empty? Please don't. I know now for each user you are creating flashcardsToLearn but im somehow one day it will change then this code will be crashing. It's not difficult to add this check and return empty list if not present ;) Or use `orElseGet` and return empty list there.